.circleci/config.yml

```diff
@@ -20,12 +19,18 @@ jobs:
                 command: 'make all'
     static-analysis:
         docker:
-            - image: debian:stretch
+            - image: circleci/buildpack-deps:bionic 
         steps:
             - checkout
             - run:
                 name: Installing cppcheck
-                command: 'apt-get install cppcheck'   
+                command: 'sudo apt-get install cppcheck'   
             - run:
                 name: Running cppcheck
-                command: 'cppcheck --error-exitcode=1 *.cpp'
+                command: 'cppcheck --error-exitcode=1 src'
+workflows:
+    version: 2
+    build_and_analyze:
+        jobs:
+            - build
+            - static-analysis
```

Comment:
Does this just run the default analysis? Do we want to customize it at all?

Reply:
It currently just runs the default analysis (which is honestly pretty comprehensive) https://sourceforge.net/p/cppcheck/wiki/ListOfChecks/ Probably fine for right now.